dianna/methods/shap.py

```diff
@@ -1,21 +1,223 @@
+        self.onnx_model = onnx.load(model)  # load onnx model
+        self.output_node = prepare(self.onnx_model, gen_tensor_dict=True).outputs[0]
+        self.input_data = input_data
+        self.channel_axis_first = channel_axis_first
+        self.background = background
+        # other keyword arguments for the method segment_image
+        max_num_iter = kwargs.get("max_num_iter", 10)
+        spacing = kwargs.get("spacing", None)
+        convert2lab = kwargs.get("convert2lab", None)
+        enforce_connectivity = kwargs.get("enforce_connectivity", True)
+        min_size_factor = kwargs.get("min_size_factor", 0.5)
+        max_size_factor = kwargs.get("max_size_factor", 3)
+        slic_zero = kwargs.get("slic_zero", False)
+        start_label = kwargs.get("start_label", 1)
+        mask = kwargs.get("mask", None)
+
+        # first check the dimension of input_data
+        if input_data.ndim != 3:
+            raise IOError(
+                "The input image must follow the required shape [height, width, color_channels] or [color_channels, height, width]"
+            )
+
+        # call the segment method to create segmentation of input image
+        self.image_segments = self._segment_image(
+            input_data,
+            n_segments,
+            compactness,
+            max_num_iter,
+            sigma,
+            spacing,
+            convert2lab,
+            enforce_connectivity,
+            min_size_factor,
+            max_size_factor,
+            slic_zero,
+            start_label,
+            mask,
+            channel_axis_first,
+        )
+
+        # call the Kernel SHAP explainer
+        explainer = shap.KernelExplainer(self._runner, np.zeros((1, n_segments)))
+
+        with warnings.catch_warnings():
+            # avoid warnings due to version conflicts
+            warnings.simplefilter("ignore")
+            shap_values = explainer.shap_values(
+                np.ones((1, n_segments)), nsamples=nsamples
+            )
+
+        return shap_values, self.image_segments
+
+    def _mask_image(
+        self, features, segmentation, image, background=None, channel_axis_first=False
+    ):
+        """Define a function that depends on a binary mask representing
+           if an image region is hidden.
+
+        Args:
+            features (np.ndarray): A matrix of samples (# samples x # features)
+                                   on which to explain the model's output.
+            segmentation (np.ndarray): Image segmentations generated by
+                                       the function _segment_image
+            image (np.ndarray): Image to be explained
+            background (int): Background color for the masked image
+            channel_axis (int): See the function explain_image
         """
-        self.model = model
+        # if the image shape is [color_channels, height, width]
+        if channel_axis_first:
+            image = np.transpose(image, (1, 2, 0))
```

Comment:
We should probably use the xarray tools here (and in other spots where the transpose is done) like we do for LIME, see here: https://github.com/dianna-ai/dianna/blob/88bcaec001e640c35e5e1e08517ef1624fd661cb/dianna/methods/lime.py#L159

Reply:
Thanks for the tip and redirect me to the functions in lime and utils @loostrum. Now the KernelSHAP module also uses the xarray and the related functions in utils to prepare the input image data, similar to what you did in LIME.